Assets/Scripts/AudioManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿using UnityEngine.Audio; 
+using UnityEngine;
+using System;
+
+public class AudioManager : MonoBehaviour
+{
+    public static AudioManager instance { get; private set; }
+    public Sound[] musicTracks;
+    public Sound[] sfxTracks;
+    
+
+    private void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+            return; 
+        }
+
+        foreach(Sound music in musicTracks)
+        {
+            music.source = gameObject.AddComponent<AudioSource>(); 
+            music.source.clip = music.audioClip;
+            music.source.loop = music.loop;
+            music.source.volume = music.volume;
+            music.source.pitch = music.pitch; 
+        }
```

Comment:
For future reference, instantiating components in runtime is expensive, but for the scope of the project, changing this up is probably unnecessary.

Reply:
We might need to set up an Audio Mixer for Music, but we can review this on a later date.

Reply:
I will keep these foreach loops for now then.